src/main/java/org/sonarlint/intellij/telemetry/TelemetryStorage.java

```diff
@@ -0,0 +1,143 @@
+  private LocalDate installDate;
+  private LocalDate lastUseDate;
+  private LocalDateTime lastUploadDateTime;
+  private Long numUseDays;
+  private boolean enabled;
+
+  public TelemetryStorage() {
+    enabled = true;
+    installDate = LocalDate.now();
+    lastUseDate = null;
+    lastUploadDateTime = null;
+    numUseDays = 0L;
+  }
+
+  public TelemetryStorage setLastUploadTime(LocalDateTime dateTime) {
+    this.lastUploadDateTime = dateTime;
+    return this;
+  }
+
+  public TelemetryStorage setEnabled(boolean enabled) {
+    this.enabled = enabled;
+    return this;
+  }
+
+  public TelemetryStorage setInstallDate(LocalDate installDate) {
+    this.installDate = installDate;
+    return this;
+  }
+
+  public TelemetryStorage setLastUseDate(LocalDate lastUseDate) {
+    this.lastUseDate = lastUseDate;
+    return this;
+  }
+
+  public TelemetryStorage setNumUseDays(Long numUseDays) {
+    this.numUseDays = numUseDays;
+    return this;
+  }
+
+  public boolean enabled() {
+    return enabled;
+  }
+
+  @CheckForNull
+  public LocalDateTime lastUploadDateTime() {
+    return lastUploadDateTime;
+  }
+
+  public LocalDate installDate() {
+    return installDate;
+  }
+
+  @CheckForNull
+  public LocalDate lastUseDate() {
+    return lastUseDate;
+  }
+
+  public Long numUseDays() {
+    return numUseDays;
+  }
+
+  public static TelemetryStorage load(Path filePath) throws IOException {
+    Gson gson = new Gson();
+    byte[] bytes = Files.readAllBytes(filePath);
+    byte[] decoded = Base64.decodeBase64(bytes);
+    String json = new String(decoded, StandardCharsets.UTF_8);
+    return validate(gson.fromJson(json, TelemetryStorage.class));
+  }
+
+  public void save(Path filePath) throws IOException {
+    Files.createDirectories(filePath.getParent());
+    Gson gson = new Gson();
+    String json = gson.toJson(this);
+    byte[] encoded = Base64.encodeBase64(json.getBytes(StandardCharsets.UTF_8));
+    Files.write(filePath, encoded);
+  }
+
+  private static TelemetryStorage validate(@Nullable TelemetryStorage telemetry) {
+    LocalDate today = LocalDate.now();
+
+    if (telemetry != null) {
+      if (telemetry.installDate() == null || telemetry.installDate().isAfter(today)) {
+        telemetry.setInstallDate(today);
+      }
+
+      LocalDate lastUseDate = telemetry.lastUseDate();
+      if (lastUseDate == null) {
+        telemetry.setNumUseDays(0L);
+      } else {
+        if (lastUseDate.isBefore(telemetry.installDate())) {
+          telemetry.setLastUseDate(telemetry.installDate());
+        } else if (lastUseDate.isAfter(today)) {
+          telemetry.setLastUseDate(today);
+        }
+
+        long maxUseDays = telemetry.installDate().until(telemetry.lastUseDate(), DAYS) + 1;
+        if (telemetry.numUseDays() > maxUseDays) {
+          telemetry.setNumUseDays(maxUseDays);
+          telemetry.setLastUseDate(today);
+        }
+      }
+    } else {
+      return new TelemetryStorage();
```

Comment:
I'd flip the if-else branches to make this an early return and the rest of the method flatter